### PR TITLE
TI-2731 - Enable users to log issues with help desk/Jira

### PR DESCRIPTION
**Pull request by stewartm83**
Please review

### CODE REVIEW

File: atlassion/atlassion.go [hunk 1]
```diff
@@ -0,0 +1,253 @@
+package atlassion
```
Comment (dbferreira) [inline atlassion/atlassion.go]: I think this should read `atlassian` (also rename folder)

File: httphandler.go [hunk 2]
```diff
@@ -44,6 +46,67 @@ func (s *MessagingServer) StartServer() error {
...
+	var statusCode int
+	if contentType == "multipart/form-data" && issueKey != "" {
+		statusCode, err = s.JiraApi.AddAttachments(issueKey, r)
+		if err != nil {
+			s.Log.Infof("Unable to add attachments to Jira ticket with Id: %v, error: %v", issueKey, err.Error())
+			http.Error(w, fmt.Sprintf("Unable to add attachments to Jira ticket with Id: %v, error: %v", issueKey, err.Error()), statusCode)
+		}
+	}
+
+	type successJson struct {
```
Comment (dbferreira) [inline httphandler.go]: I think it's standard practice to define the types at the start of the module file (I could be wrong).

**Review by dbferreira — CHANGES_REQUESTED** [review]

File: httphandler.go [hunk 3]
```diff
@@ -44,6 +51,66 @@ func (s *MessagingServer) StartServer() error {
...
+	Server: %v
+	
+	Message:					
+	%v
+	`
+	message := fmt.Sprintf(messageTemplate, clientName, clientEmail, clientPhoneNumber, callback, browserName, browserVersion, serverAddress, description)
+
+	issueKey, err := s.JiraApi.CreateIssue(summary, message)
+	if err != nil {
+		s.Log.Infof("Unable to log issue with the Jira Api: %v", err.Error())
```
Comment (RoanBrand) [inline httphandler.go]: Change log level to warn or error

File: httphandler.go [hunk 4]
```diff
@@ -44,6 +51,66 @@ func (s *MessagingServer) StartServer() error {
...
+		if err != nil {
+			s.Log.Infof("Unable to add attachments to Jira ticket with Id: %v, error: %v", issueKey, err.Error())
+			http.Error(w, fmt.Sprintf("Unable to add attachments to Jira ticket with Id: %v, error: %v", issueKey, err.Error()), statusCode)
+			return
+		}
+	}
+
+	result := successJson{key: issueKey, code: statusCode}
+
+	resultJson, err := json.Marshal(&result)
```
Comment (RoanBrand) [inline httphandler.go]: Rather:

```
enc := json.NewEncoder(w)
w.Header().Set("Content-Type", "application/json")
err = enc.Encode(&result)
if err != nil {
    ...
}
```

`json.Marshal` allocates memory by creating a byte slice. using the encoder you will achieve the same result but spare resources.

File: httphandler.go [hunk 5]
```diff
@@ -44,6 +51,66 @@ func (s *MessagingServer) StartServer() error {
...
+		http.Error(w, fmt.Sprintf("Unable to log issue with the Jira Api: %v", err.Error()), http.StatusInternalServerError)
+		return
+	}
+
+	ct := r.Header.Get("Content-Type")
+	contentTypeList := strings.Split(ct, ";")
+	contentType := contentTypeList[0]
+
+	var statusCode int
+	if contentType == "multipart/form-data" && issueKey != "" {
```
Comment (RoanBrand) [inline httphandler.go]: what if this `if` isn't taken? `statusCode` would be 0 and still used

**Review by RoanBrand — CHANGES_REQUESTED** [review]

**Pull request merged**